src/core/services/mongo.service.ts

```diff
@@ -0,0 +1,243 @@
+import mongodb from 'mongodb';
+import { injectable } from "inversify";
+import assert from 'assert';
+
+import container from '../../inversity.config';
+
+import ConfigService from './config.service';
+import LoggerService from './logger.service';
+
+@injectable()
+export default class MongoService {
+  private _configService: ConfigService = container.resolve<ConfigService>(ConfigService);
+  private _loggerService: LoggerService = container.resolve<LoggerService>(LoggerService);
+
+  private _mongoClient: mongodb.MongoClient;
+  public _db: mongodb.Db;
+
+  public async connect(): Promise<void> {
+    return new Promise((resolve, reject) => {
+      this._mongoClient = new mongodb.MongoClient(this._configService.get('MONGODB_URI'), { useUnifiedTopology: true });
+
+      this._mongoClient.connect((err) => {
+        if (err) {
+          this._loggerService.log('error', 'Venom could not connect to MongoDB', err);
+          reject();
+        } else {
+          this._loggerService.log('info', 'Venom is connected to MongoDB');
+
+          this._db = this._mongoClient.db(this._configService.get('MONGODB_DB_NAME'));
+
+          resolve();
+        }
+      });
+    });
+  }
+
+  public disconnect() {
+    if (this._mongoClient && this._mongoClient.isConnected) {
+      this._loggerService.log('info', 'Closing connection to MongoDB');
+      this._mongoClient.close();
+    }
+  }
+
+  /**
+   * Fetches the first document that matches the query
+   *
+   * @param collection String identifier for collection document belongs to
+   * @param query key/value pairs of search conditions
+   *
+   * @returns Returns matched document
+   *
+   * @example findOne('123456', 'collectionName', { ident: 'generated-slug' })
+   */
+  public async findOne(userId: string, collection: string, query: any) {
```

Comment:
Where do we use userId? Why is it needed here? I believe user permissions should be on a higher level, somewhere, if that was the purpose of this.

Reply:
It's for the logging library, so that there is a userId attached to each database command that happens as well. Down the line I'd prefer to separate it into an audit system, separate from the logger, probably attached to commands.